src/vs/workbench/parts/terminal/electron-browser/windowsShellHelper.ts

```diff
@@ -89,7 +98,18 @@ export class WindowsShellHelper {
 			// Save the pid in the stack and keep looking for children of that child
 			this._childProcessIdStack.push(result[0].pid);
 			return this.refreshShellProcessTree(result[0].pid, result[0].executable);
-		}, error => { return error; });
+		}, error => {
+			if (!this._isDisposed) {
+				return error;
```

Comment:
This shouldn't be returning `error`, because it's probably an `Error` object, and this method is trying to return a `string`. Have you tested what happens if `getChildProcessDetails` fails for some reason and we return null or the error?